src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Attributes/JsonRequiredAttribute.cs

```diff
@@ -1,22 +1,24 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Text.Json.Serialization.Metadata;
+
 namespace System.Text.Json.Serialization
 {
     /// <summary>
-    /// Specifies the property is required when deserializing.
+    /// Indicates that the annotated member must bind to a JSON property on deserialization.
     /// </summary>
+    /// <remarks>
+    /// <see langword="null"/> token in JSON will not trigger a validation error.
+    /// For contracts originating from <see cref="DefaultJsonTypeInfoResolver"/> or <see cref="JsonSerializerContext"/>,
+    /// this attribute will be mapped to <see cref="JsonPropertyInfo.IsRequired"/>.
+    /// </remarks>
     [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field, AllowMultiple = false)]
```

Comment:
Should this also support `AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Interface` for blanket application of `IsRequired` to members? Does Json.NET support something similar?

Reply:
AFAIK not possible with Json.NET because they only have JsonPropertyAttribute with Required property and that can only be put on property/field/parameter

Reply:
We can revisit this behavior in the future since it won't be breaking to add such support or discuss that during API review. For now I don't believe that will be super useful (also user can easily define their own custom attribute to achieve that)

Reply:
Sounds good. If demand arises we can always retrofit the attribute in the future.